### PR TITLE
raising http errors

### PR DESCRIPTION
**Pull request by BoltzmannBrain**

### CODE REVIEW

**Comment by BoltzmannBrain** [comment]
@rhyolight please review.

**Comment by BoltzmannBrain** [comment]
@scottpurdy please review.

**Comment by scottpurdy** [comment]
It might be better to make the errors more specific to your API. If the user tries to encode a word and you get an HTTP error code because cortical.io doesn't have a representation for it then "UnknownEncodingError" or something like that is probably more descriptive than something to do with HTTP codes (which is an implementation detail that the user probably doesn't care about).

**Comment by BoltzmannBrain** [comment]
@scottpurdy better?

File: cortipy/cortical_client.py [hunk 1]
```diff
@@ -153,16 +154,16 @@ def _queryAPI(self, method, resourcePath, queryParams,
       response = requests.post(
         url, params=queryParams, headers=headers, data=postData)
     else:
-      raise Exception('Method ' + method + ' is not recognized.')
+      raise RequestMethodError('Method ' + method + ' is not recognized.')
```
Comment (scottpurdy) [inline cortipy/cortical_client.py]: double quotes

File: cortipy/cortical_client.py [hunk 2]
```diff
@@ -153,16 +154,16 @@ def _queryAPI(self, method, resourcePath, queryParams,
...
     if response.status_code != 200:
-      raise Exception("Response " + str(response.status_code)
-                      + ": " + response.content)
+      raise UnsuccessfulEncodingError("Response " + str(response.status_code)
+                               + ": " + response.content)
```
Comment (scottpurdy) [inline cortipy/cortical_client.py]: fix indentation. And when line breaking on an operator it is more common to leave the operator on the previous line so it is obvious that the statement continues to the next line

File: tests/unit/client_test.py [hunk 3]
```diff
@@ -94,7 +94,43 @@ def testPostQueryToAPI(self, mockPost):
     # Assert:
     self.assertEqual({"dummy": "mock body"}, response)
   
-  
+
+  @patch.object(requests, 'post')
```
Comment (scottpurdy) [inline tests/unit/client_test.py]: double quotes

File: tests/unit/client_test.py [hunk 4]
```diff
@@ -94,7 +94,43 @@ def testPostQueryToAPI(self, mockPost):
...
+    Tests the client sending an invalid query method to the API, asserting we
+    receive a RequestMethodError.
+    """
+    mockPost.return_value = Mock(
+      content='{"dummy": "mock body"}', status_code=None)
+    
+    # Act:
+    client = cortipy.CorticalClient(apiKey="fakeKey")
+    try:
+      response = client._queryAPI("BAD_METHOD", "path", {})
```
Comment (scottpurdy) [inline tests/unit/client_test.py]: use `with self.assertRaises(ErrorType) as e:` to validate the exceptions

**Comment by scottpurdy** [comment]
done reviewing

**Comment by BoltzmannBrain** [comment]
@scottpurdy thank you for the reviews. Should be all set now.

File: cortipy/exceptions.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""Exceptions for querying the Cortical.io API."""
```
Comment (scottpurdy) [inline cortipy/exceptions.py:22]: Might just be explicit that these are cortipy exceptions (Cortical.io API kind of sounds like you're talking about the HTTP API)

File: tests/unit/client_test.py [hunk 6]
```diff
@@ -94,7 +95,41 @@ def testPostQueryToAPI(self, mockPost):
...
+    """
+    mockPost.return_value = Mock(
+      content='{"dummy": "mock body"}', status_code=None)
+    
+    # Act:
+    client = cortipy.CorticalClient(apiKey="fakeKey")
+
+    # Assert:
+    with self.assertRaises(RequestMethodError):
+      response = client._queryAPI("BAD_METHOD", "path", {})
```
Comment (scottpurdy) [inline tests/unit/client_test.py]: don't store the response if you aren't using it

**Comment by scottpurdy** [comment]
:+1: merge when ready

**Pull request merged**